### PR TITLE
METRON-1290: Only first 10 alerts are update when a MetaAlert status is changed to inactive

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
This PR fixes a small bug in the ElasticsearchMetaAlertDao that only updates the first 10 alerts in a metaalert when that metaalert status is changed to inactive.  This is due to the fact that the ElasticsearchMetaAlertDao needs to lookup up the child alerts and a search created with `elasticsearchDao.getClient().prepareSearch()` defaults to result set size to 10.  This can be reproduced in full dev:
- turn off the sensor stubs so that the number of alerts in ES is fixed
- take note of the number of alerts in the index
- create a metaalert with more than 10 alerts
- the number of alerts should decrease to: # of alerts added - 1 metaalert
- change the metaalert status to inactive 
- the number of alerts will now be less than before where the count should be the same

I also added an embarrassingly complex unit test to cover this situation.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by merrimanr** [comment]
Looks like there is a [bug](https://stackoverflow.com/questions/40984882/mockito-and-powermock-methodnotfoundexception-being-thrown) in the powermock version inherited by this module.  Latest commit fixes that (and hopefully helps others in the future that want to use powermock in Metron).

**Comment by nickwallen** [comment]
Are we able to replicate this defect in the integration tests?  Personally, I'd prefer to see the integration tests validate something like this, if at all possible.

File: metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/dao/ElasticsearchMetaAlertDaoTest.java [hunk 1]
```diff
@@ -489,4 +505,54 @@ public void testHandleMetaUpdateAlert() throws IOException {
     verify(mockEsDao, times(1)).getLatest(guidAlert, null);
     verify(mockEsDao, times(1)).batchUpdate(updates);
   }
+
+  @Test
+  public void getAllAlertsForMetaAlertShouldReturnAllAlerts() throws Exception {
```
Comment (nickwallen) [inline metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/dao/ElasticsearchMetaAlertDaoTest.java]: Are we able to replicate this defect in the integration tests? Personally, I'd prefer to see the integration tests validate something like this, if at all possible.  That would also help our migration to ES 5.x over this mock driven test.

**Comment by iraghumitra** [comment]
+1 for functionality, tested on full dev when status is changed on a meta-alert all the alerts in the meta-alert are updated with the correct status

**Comment by justinleet** [comment]
@merrimanr This is already taken care of by the various refactoring in https://github.com/apache/metron/pull/824, right? Can you close this (and the associated jira) if that's accurate?

**Comment by nickwallen** [comment]
Even if the functionality is addressed, are there tests that cover this condition that we can pull in from this PR?  I don't think we ever got a proper integration test to cover this issue, but maybe @merrimanr had created one on his private branch?

**Comment by justinleet** [comment]
@nickwallen I'm about to open a PR with an updated integration test.  We should be able to close this one, since the changes here aren't relevant anymore.

**Comment by justinleet** [comment]
@nickwallen https://github.com/apache/metron/pull/842
@merrimanr If you're good with that PR, feel free to close this one.

**Comment by nickwallen** [comment]
Sounds good.  I will take a look at #842.

**Pull request closed without merging**